### PR TITLE
chore: Added docs for extension feature

### PR DESCRIPTION
**Pull request by abhilash-kumar-nair**
Docs for the extension feature
@jacobtaxen

### CODE REVIEW

File: docs/source/tutorial.rst [hunk 1]
```diff
@@ -145,17 +145,71 @@ They can be set in a way similar to the compiler_options::
...
+A simpler approach for parameterization also exists for scenarios where only variable modifiers are varied for setting up
+multi-execution cases. This could be done by calling the ``with_cases()`` method on the ``experiment_definition`` class object
+with the variable modifiers as inputs::
+   
+   experiment_definition = experiment_definition.with_cases([{'PI.k': 20}, {'PI.k': 30}])
+
+**Note:**
+
+It is not supported to have both range operator and experiment extensions defined for an experiment. The simulation cases
+could only be setup with one of the two methods. However, it is allowed to call the  ``with_modifiers`` method on the
```
Comment (jacobtaxen) [inline docs/source/tutorial.rst:208]: setup -> set up

File: docs/source/tutorial.rst [hunk 2]
```diff
@@ -145,17 +145,71 @@ They can be set in a way similar to the compiler_options::
...
+   
+   experiment_definition = experiment_definition.with_cases([{'PI.k': 20}, {'PI.k': 30}])
+
+**Note:**
+
+It is not supported to have both range operator and experiment extensions defined for an experiment. The simulation cases
+could only be setup with one of the two methods. However, it is allowed to call the  ``with_modifiers`` method on the 
+``experiment_definition`` class to specify variable to modify. The modified variable in such a scenario would be set in all
+the cases defined using the ``with_extensions()`` or ``with_cases()`` method calls. If the same variable modifier is 
+set in both ``experiment_definition`` and extensions, the one set in the extensions would gain precedence override the former.
```
Comment (jacobtaxen) [inline docs/source/tutorial.rst:211]: gain precedence override the former -> gain precedence, overriding the former

File: docs/source/tutorial.rst [hunk 3]
```diff
@@ -145,17 +145,71 @@ They can be set in a way similar to the compiler_options::
 Setting up a series of simulations
 ----------------------------------
 
-The ``experiment_definition`` created could be modified to set up a series of simulations by specifying the
-the parameter to sweep and the range by calling the ``with_modifiers()`` method on the ``experiment_definition``::
+The ``experiment_definition`` created could be modified to set up a series of simulations following either of 
+approaches listed below:
```
Comment (jacobtaxen) [inline docs/source/tutorial.rst:149]: this sentence feels a bit weird, maybe "Following either of the approaches listed below, the created experiment definition can be modified to set up a series of simulations" or something

**Review by efredriksson-modelon — CHANGES_REQUESTED** [review]
Nice 👍 Just a comment about how much focus we should put on the specific range operator.

File: docs/source/quick_start.rst [hunk 4]
```diff
@@ -70,6 +70,27 @@ An example of setting up and executing a series of simulations on a server and r
    plt.grid()
    plt.show()
 
+A more flexible and customized way to define a series of simulations::
```
[the remark this answers is not in the document]
Comment (abhilash-kumar-nair) [reply docs/source/quick_start.rst:73]: The two colons signify a code block below :)

**Review by efredriksson-modelon — APPROVED** [review]
👍

**Review by jacobtaxen — APPROVED** [review]
Looks good!

**Pull request merged**